libs/gl-plugin/src/node/mod.rs

```diff
@@ -114,10 +106,11 @@ impl PluginNodeServer {
             use tokio::time::{sleep, Duration};
 
             // Move the lock into the closure so we can release it later.
-            let rpc = rrpc.lock().await;
+            let mut rpc = cln_rpc::ClnRpc::new(rpc_path.clone()).await.unwrap();
```

Comment:
We got a bit unlucky here: we use the `Mutex` to prevent incoming JSON-RPC calls, before the JSON-RPC socket is ready. This is done by acquiring the lock during startup and releasing it once the socket file is present. The internal use-cases are ok, since they by definition are invoked only after the node started and the socket is ready.

The change below however breaks this, by not locking in `get_rpc` it is now possible for incoming grpc calls to be directly forwarded (not waiting behind a lock) and so they will fail. We will most likely notice this as the startup commands (i.e., the ones that caused the node to start) will fail in a large number of cases.

Please leave the `Mutex` in `get_rpc`

Reply:
I moved the rpc client into a `Mutex` inside of a `OnceCell` for a clean  one-time initialization.